tests/JITBenchmark.java

```diff
@@ -30,32 +38,154 @@ public static void createPrimitiveArrays() {
-    for (int i = 0; i < size * 8; i++) {
+    int count = size * 16;
+    for (int i = 0; i < count; i++) {
       stream.write(i);
       stream.write(i);
       stream.write(i);
       stream.write(i);
     }
+  }
 
-    System.out.println("Length: " + stream.size());
+  public static void concatStrings() {
+    String s = "";
+    int count = size / 16;
+    for (int i = 0; i < count; i++) {
+      s += "X";
+    }
   }
 
-  public static void main(String[] args) {
-    long start = 0;
+  public static void getBytes() {
+    String s = "getBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytesgetBytes";
+    int count = size / 4;
+    byte [] bytes = null;
+    for (int i = 0; i < count; i++) {
+      bytes = s.getBytes();
+    }
+  }
 
-    start = System.currentTimeMillis();
-    System.out.println("createObjectArrays: ...");
-    // createObjectArrays();
-    System.out.println("createObjectArrays: " + (System.currentTimeMillis() - start));
+  public static void synch() {
+    int count = size * 32;
+    byte [] bytes = null;
+    Object o = new Object();
+    int sum = 0;
+    for (int i = 0; i < count; i++) {
+      synchronized (o) {
+        sum++;
+      }
+      synchronized (o) {
+        sum++;
+      }
+      synchronized (o) {
+        sum++;
+      }
+      synchronized (o) {
+        sum++;
+      }
+      synchronized (o) {
+        sum++;
+      }
+      synchronized (o) {
+        sum++;
+      }
+    }
+  }
 
+  public static void hashtable() {
+    int count = size * 2;
+    Hashtable hash = new Hashtable();
+    Object o = new Object();
+    String [] names = {
+      "hello", "world", "hello1", "world2", "hello3", "world4", "hello5", "world6"
+    };
+    for (int i = 0; i < count; i++) {
+      String name = names[i % names.length];
+      hash.put(name, o);
+      hash.get(name);
+      hash.put(name, o);
+      hash.get(name);
+      hash.put(name, o);
+      hash.get(name);
+      hash.put(name, o);
+      hash.get(name);
+    }
+  }
+
+  public static void arrayTypeCheck() {
+    A [] array = new A [1024];
+    A a = new A();
+    B b = new B();
+    int count = size * 3;
+    for (int i = 0; i < count - 1; i++) {
+      array[i % 1024] = a;
+      array[i % 1024] = b;
+    }
+  }
+
+  public static void begin() {
+    System.gc();
     start = System.currentTimeMillis();
-    System.out.println("createPrimitiveArrays: ...");
-    // createPrimitiveArrays();
-    System.out.println("createPrimitiveArrays: " + (System.currentTimeMillis() - start));
+  }
 
+  public static void finish(String name) {
+    System.out.println(name + ": " + (System.currentTimeMillis() - start));
+  }
+
+  public static void main(String[] args) {
+
+    // Sys.eval("J2ME.emitCheckArrayStore = false;");
+    // Sys.eval("J2ME.emitCheckArrayBounds = false;");
+
+    size = 1024;
+
+    begin();
     start = System.currentTimeMillis();
```

Comment:
We should use JVM.monotonicTimeMillis() that uses performance.now() instead of Date.now()